src/Microsoft.AspNet.Http/DefaultHttpResponse.cs

```diff
@@ -54,21 +56,20 @@ public override Stream Body
 
         public override long? ContentLength
         {
-            get
-            {
-                return ParsingHelpers.GetContentLength(Headers);
-            }
-            set
-            {
-                ParsingHelpers.SetContentLength(Headers, value);
-            }
+            get { return HttpResponseFeature.ContentLength; }
```

Comment:
Isn't this broken? Don't you still have to support setting the header? Can this and the header get out of sync?